tests/unittests/core/conftest.py

```diff
@@ -155,53 +156,160 @@ def random_dt(monkeypatch):
+        "params": [
+            {"name": "/decoding_layer", "type": "categorical", "value": "rnn"},
+            {"name": "/encoding_layer", "type": "categorical", "value": "gru"},
+        ],
+        "parents": [],
+    },
+    {
+        "experiment": "supernaedo2-dendi",
+        "status": "new",
+        "worker": None,
+        "submit_time": MockDatetime(2017, 11, 23, 2),
+        "start_time": None,
+        "end_time": None,
+        "results": [{"name": None, "type": "objective", "value": None}],
+        "params": [
+            {
+                "name": "/decoding_layer",
+                "type": "categorical",
+                "value": "lstm_with_attention",
+            },
+            {"name": "/encoding_layer", "type": "categorical", "value": "rnn"},
+        ],
+        "parents": [],
+    },
+    {
+        "experiment": "supernaedo2-dendi",
+        "status": "interrupted",
+        "worker": None,
+        "submit_time": MockDatetime(2017, 11, 23, 3),
+        "start_time": MockDatetime(2017, 11, 23, 3),
+        "end_time": None,
+        "results": [{"name": None, "type": "objective", "value": None}],
+        "params": [
+            {
+                "name": "/decoding_layer",
+                "type": "categorical",
+                "value": "lstm_with_attention",
+            },
+            {"name": "/encoding_layer", "type": "categorical", "value": "lstm"},
+        ],
+        "parents": [],
+    },
+    {
+        "experiment": "supernaedo2-dendi",
+        "status": "suspended",
+        "worker": None,
+        "submit_time": MockDatetime(2017, 11, 23, 4),
+        "start_time": MockDatetime(2017, 11, 23, 4),
+        "end_time": None,
+        "results": [{"name": None, "type": "objective", "value": None}],
+        "params": [
+            {"name": "/decoding_layer", "type": "categorical", "value": "gru"},
+            {"name": "/encoding_layer", "type": "categorical", "value": "lstm"},
+        ],
+        "parents": [],
+    },
+]
 
 
 @pytest.fixture()
-def refers_id_substitution(with_user_tsirif, random_dt, clean_db, create_db_instance):
-    """Replace trial ids by the actual ids of the experiments."""
-    db = create_db_instance
-    query = {"metadata.user": "tsirif"}
-    selection = {"name": 1, "refers": 1}
-    experiments = db.read("experiments", query, selection)
-    experiment_dict = dict(
-        (experiment["name"], experiment) for experiment in experiments
-    )
-
-    for experiment in experiments:
-        query = {"_id": experiment["_id"]}
-
-        root_id = experiment_dict[experiment["refers"]["root_id"]]["_id"]
-        if experiment["refers"]["parent_id"] is not None:
-            parent_id = experiment_dict[experiment["refers"]["parent_id"]]["_id"]
-        else:
-            parent_id = None
-        update = {"refers.root_id": root_id, "refers.parent_id": parent_id}
-        db.write("experiments", update, query)
+def hacked_exp(with_user_dendi, random_dt, storage):
```

Comment:
Ok I see what the issue is. You do not need `storage` if you use the contextmanager OrionState because the latter is creating a new storage object. But since you are returning the exp in this fixture, you don't have access directly to the storage in the tests. A solution would be to avoid OrionState and do `storage.create_experiment(dendi_exp_config)` with `storage._db.write('trials', dendi_base_trials)`.

Reply:
Ohh, okay, that makes sense. I will do what you describe so that the fixtures are coherent. Thanks!